crates/driver/src/infra/persistence/mod.rs

```diff
@@ -0,0 +1,77 @@
+mod dto;
+
+use {
+    crate::{
+        domain::{
+            competition::{auction::Id, Auction},
+            liquidity,
+        },
+        infra::solver::Config,
+    },
+    serde::Deserialize,
+    std::sync::Arc,
+    tracing::Instrument,
+};
+
+#[derive(Clone, Debug, Default, Deserialize)]
+#[serde(rename_all = "kebab-case", deny_unknown_fields)]
+pub struct S3 {
+    /// The s3_instance_upload_* arguments configure how auction instances
+    /// should be uploaded to AWS S3.
+    /// They must either all be set or all not set.
+    pub s3_instance_upload_bucket: String,
+
+    /// Prepended to the auction id to form the final instance filename on S3.
+    /// Something like "staging/mainnet/"
+    pub s3_instance_upload_filename_prefix: String,
+}
+
+impl From<S3> for s3::Config {
+    fn from(value: S3) -> Self {
+        Self {
+            bucket: value.s3_instance_upload_bucket,
+            filename_prefix: value.s3_instance_upload_filename_prefix,
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Persistence {
+    s3: Option<Arc<s3::Uploader>>,
+}
+
+impl Persistence {
+    pub async fn build(config: &Config) -> Self {
+        if let Some(s3) = &config.s3 {
+            Self {
+                s3: Some(Arc::new(s3::Uploader::new(s3.clone().into()).await)),
+            }
+        } else {
+            Self { s3: None }
+        }
+    }
+
+    /// Saves the given auction with liquidity
```

Comment:
nanonit: document fire and forget mentality (so the caller doesn't think this may be blocking an upload of a large file